antarest/study/storage/variantstudy/variant_study_service.py

```diff
@@ -1050,3 +1051,35 @@ def initialize_additional_data(self, variant_study: VariantStudy) -> bool:
                 exc_info=e,
             )
             return False
+
+    def clear_all_snapshots(self, limit: int, params: t.Optional[RequestParameters] = None) -> None:
+        """
+        Clear all variant snapshots older than `limit` (in hours).
+        Only available for admin users.
+        `limit` must be a positive integer.
+        Raises a UserHasNotPermissionError if the user is not administrator
+        Args:
+            limit (integer): number of hours
+            params: request parameters used to identify the user status
+        Returns: None
+
+        Raises:
+            UserHasNotPermissionError
+            HTTPException
+        """
+        if limit < 0:
+            raise HTTPException(status_code=400, detail=f"Limit cannot be negative (limit={limit})")
+        if params.user.is_site_admin():
+            result = self.repository.get_all(
+                study_filter=StudyFilter(
+                    variant=True,
+                    access_permissions=AccessPermissions(is_admin=True),
+                )
+            )
+            for variant in result:
+                if variant.updated_at < datetime.now() - timedelta(hours=limit) or (
+                    variant.last_access and variant.last_access < datetime.now() - timedelta(hours=limit)
+                ):
+                    self.clear_snapshot(variant)
+        else:
+            raise UserHasNotPermissionError
```

Comment:
You should raise an instance of the class. I am not sure but that should be `UserHasNotPermissionError()`.